### PR TITLE
Convert CategoryPillInner into functional component

### PR DESCRIPTION
**Pull request by themakshter**

### CODE REVIEW

**Comment by themakshter** [comment]
This one was a bit more complex, so I might have made some errors in converting. Let me know if something doesn't make sense.

I did test updating the category of a question and it is working (I used that to fix a bug before my commit)

**Comment by themakshter** [comment]
Update: I moved the extraction of the function from outside back to inside because I think it's clearer in this PR. I will move it outside in a follow-up change.

**Review by drimpact — APPROVED** [review]
Thanks!

File: src/app/components/CategoryPill/index.tsx [hunk 1]
```diff
@@ -107,101 +96,90 @@ class CategoryPillInner extends React.Component<IProps, IState> {
...
+  }
+  const cat = getCategory(q.categoryID);
+  if (cat === null || cat === undefined) {
+    return renderPill("empty", t("Unknown Category"));
+  }
+  return renderPill("set", cat.name);
+};
 
 const CategoryPillConnected = setCategory<IProps>(CategoryPillInner);
 const CategoryPill = withTranslation()(CategoryPillConnected);
```
[the remark this answers is not in the document]
Comment (themakshter) [reply src/app/components/CategoryPill/index.tsx]: Updated

**Review by drimpact — APPROVED** [review]

**Pull request merged**